core/src/main/scala/fs2/task.scala

```diff
@@ -202,14 +202,14 @@ object Task extends Instances {
    * stack overflows.
    */
   def suspend[A](a: => Task[A]): Task[A] = new Task(Future.suspend(
-    Try(a.get) match {
+    Attempt(a.get) match {
       case Left(e) => Future.now(Left(e))
       case Right(f) => f
   }))
 
   /** Create a `Future` that will evaluate `a` using the given `Strategy`. */
   def apply[A](a: => A)(implicit S: Strategy): Task[A] =
-    async { cb => S(cb(Try(a))) }
+    async { cb => S(cb(Attempt(a))) }
```

Comment:
I'm fine with adding `success` and `failure` but I don't think these replace `Attempt.apply` -- rather, they complement.